src/main/java/net/helpscout/api/ApiClient.java

```diff
@@ -1442,52 +1429,23 @@ private void setThreadProperties(ConversationThread thread) {
 
     private <T> Page<T> getPage(String url, Map<String,String> params, Class<T> clazzType, int expectedCode) throws ApiException {
         url = setParams(url, params);
-        String json = doGet(url, HTTP_STATUS_OK);
+        String json = httpMethodWrapper.doGet(url, HTTP_STATUS_OK);
         JsonElement obj = parseJson(url, json);
         
         return JSONUtils.objectToPage(obj.getAsJsonObject(), clazzType);
     }
-    
-    private <T> T doPost(String url, String requestBody, int expectedCode, ResultExtractor<T> extractor) throws ApiException {
-
-        try (HTTPConnectionWrapper conn = new HTTPConnectionWrapper(apiKey, baseUrl + url, METHOD_POST, expectedCode, requestBody)) {
-            return extractor.extract(conn);
-        } catch(ApiException ex) {
-            throw ex;
-        } catch (Exception ex) {
-            throw new RuntimeException(ex);
-        }
-    }
-
-    private void doPut(String url, String requestBody, int expectedCode) throws ApiException {
 
-        try (HTTPConnectionWrapper conn = new HTTPConnectionWrapper(apiKey, baseUrl + url, METHOD_PUT, expectedCode, requestBody)) {
-        } catch(ApiException ex) {
-            throw ex;
-        } catch (Exception ex) {
-            throw new RuntimeException(ex);
-        }
-    }
+    private <T> ArrayList<T> getPageItems(String url, Class<T> clazzType) throws ApiException {
+        String json = httpMethodWrapper.doGet(url, HTTP_STATUS_OK);
+        JsonElement obj = parseJson(url, json);
 
-    private String doGet(String url, int expectedCode) throws ApiException {
-        String response    = null;
-        try (HTTPConnectionWrapper conn = new HTTPConnectionWrapper(apiKey, baseUrl + url, METHOD_GET, expectedCode)) {
-            response = conn.getResponse();
-        } catch(ApiException e) {
-            throw e;
-        } catch(Exception e) {
-            throw new RuntimeException(e);
-        }
-        return response;
+        return JSONUtils.getPageItems(obj, clazzType);
     }
 
-    private void doDelete(String url, int expectedCode) throws ApiException {
-        try (HTTPConnectionWrapper conn = new HTTPConnectionWrapper(apiKey, baseUrl + url, METHOD_DELETE, expectedCode)) {
-        } catch(ApiException e) {
-            throw e;
-        } catch (Exception ex) {
-            throw new RuntimeException(ex);
-        }
+    private JsonElement parseJson(String url, String json) {
+        LoggerFactory.getLogger(getClass()).trace("{}: {}", url, json);
+        JsonElement obj = (new JsonParser()).parse(json);
+        return obj;
```

Comment:
I think you can return result raight away:
```
return new JsonParser().parse(json);
```

Also, does the `JsonParser` need to be created every time?